aea/helpers/dialogue/base.py

```diff
@@ -830,106 +849,88 @@ def update(self, message: Message) -> Optional[Dialogue]:
-            dialogue_reference[0] != Dialogue.OPPONENT_STARTER_REFERENCE
-            and dialogue_reference[1] == Dialogue.OPPONENT_STARTER_REFERENCE
-            and message.message_id > 1
         )
+
         if is_invalid_label:
             dialogue = None  # type: Optional[Dialogue]
-        elif is_new_dialogue and message.is_incoming:  # new dialogue by other
+        elif is_new_dialogue:  # initial message for new dialogue
             dialogue = self._create_opponent_initiated(
-                dialogue_opponent_addr=message.counterparty,
+                dialogue_opponent_addr=message.sender,
                 dialogue_reference=dialogue_reference,
                 role=self._role_from_first_message(message),
             )
-        elif is_new_dialogue and not message.is_incoming:  # new dialogue by self
-            dialogue = self._create_self_initiated(
-                dialogue_opponent_addr=message.counterparty,
-                dialogue_reference=dialogue_reference,
-                role=self._role_from_first_message(message),
-            )
-        elif (  # non-initial message with incomplete label
-            is_incomplete_label_and_non_initial_msg
-        ):
-            # we can allow a dialogue to have incomplete reference
-            # as multiple messages can be sent before one is received with complete reference
-            dialogue = self.get_dialogue(message)
-        else:
-            self._update_self_initiated_dialogue_label_on_message_with_complete_reference(
-                message
-            )
+        else:  # non-initial message for existing dialogue
+            self._complete_dialogue_reference(message)
             dialogue = self.get_dialogue(message)
 
-        if dialogue is not None and dialogue.update(message):
-            result = dialogue  # type: Optional[Dialogue]
-        else:  # couldn't find the dialogue or invalid message
+        if dialogue is not None:
+            try:
+                dialogue.update(message)
+                result = dialogue  # type: Optional[Dialogue]
+            except InvalidDialogueMessage:
+                # invalid message for the dialogue found
+                result = None
+        else:
+            # couldn't find the dialogue referenced by the message
             result = None
 
         return result
 
-    def _update_self_initiated_dialogue_label_on_message_with_complete_reference(
-        self, message: Message
-    ) -> None:
+    def _complete_dialogue_reference(self, message: Message) -> None:
         """
         Update a self initiated dialogue label with a complete dialogue reference from counterparty's first message.
 
         :param message: A message in the dialogue (the first by the counterparty with a complete reference)
         :return: None
         """
-        dialogue_reference = message.dialogue_reference
+        complete_dialogue_reference = message.dialogue_reference
```

Comment:
currently wrong var name; the `complete_dialogue_reference` can currently be complete or incomplete; once the change above (line 868 onwards) is reverted it will be by default complete

Reply:
See above